spec/controllers/admin/products_controller_spec.rb

```diff
@@ -0,0 +1,114 @@
+require "rails_helper"
+include SessionsHelper
+RSpec.describe Admin::ProductsController, type: :controller do
+  let(:user_test){FactoryBot.create :user, role: 1}
+  let!(:product_test){FactoryBot.create :product}
+
+  before do
+    log_in user_test
+  end
```

Comment:
before sẽ chạy như thế nào em nhỉ

Reply:
đây em dùng before each để login vào user admin. before each thì chạy trước mỗi test case ở block it